samples/dotnet/kernel-syntax-examples/Example12_Planning.cs

```diff
@@ -19,10 +19,50 @@ internal static class Example12_Planning
 {
     public static async Task RunAsync()
     {
-        await PoetrySamplesAsync();
-        await EmailSamplesAsync();
-        await BookSamplesAsync();
-        await MemorySampleAsync();
+        //await PoetrySamplesAsync();
+        //await EmailSamplesAsync();
+        //await BookSamplesAsync();
+        //await MemorySampleAsync();
+        await ConditionalSampleAsync();
+    }
+
+    private static async Task ConditionalSampleAsync()
+    {
+        Console.WriteLine("======== Planning - Conditional flow example ========");
+        var kernel = InitializeKernelAndPlanner(out var planner);
+
+        // Load additional skills to enable planner to do non-trivial asks.
+        string folder = RepoFiles.SampleSkillsPath();
+        kernel.ImportSemanticSkillFromDirectory(folder, "FunSkill");
+        kernel.ImportSemanticSkillFromDirectory(folder, "WriterSkill");
+        kernel.ImportSkill(new TimeSkill());
+
+        var originalPlan = await kernel.RunAsync(
+            @"If is still morning please give me a joke about coffee otherwise tell me one about afternoon, but if its night give me a poem about the moon",
+            planner["CreatePlan"]);
+
+        /*
+        <goal>If is still morning please give me a joke about coffee otherwise tell me one about afternoon, but if its night give me a poem about the moon</goal>
+        <plan>
+          <function._GLOBAL_FUNCTIONS_.HourNumber setContextVariable="CURRENT_HOUR"/>
+          <if condition="$CURRENT_HOUR lessthan 12">
+            <function.FunSkill.Joke input="coffee"/>
+          </if>
+          <else>
+            <if condition="$CURRENT_HOUR lessthan 18">
+              <function.FunSkill.Joke input="afternoon"/>
+            </if>
+            <else>
+              <function.FunSkill.ShortPoem input="the moon"/>
+            </else>
+          </else>
+        </plan>
+        */
+
+        Console.WriteLine("Original plan:");
+        Console.WriteLine(originalPlan.Variables.ToPlan().PlanString);
+
+        await ExecutePlanAsync(kernel, planner, originalPlan, 20);
```

Comment:
I'm still curious, will the plan you've shown take 2 or 3 execute calls to complete when less than 12?

Reply:
Currently I do two calls to evalueate a condition.
1. Validate If statement and get usedvariables.
2. Evaluate Condition

Reply:
```
  <plan>
    <if condition="$CURRENT_HOUR lessthan 12">
      <function.FunSkill.Joke input="coffee"/>
    </if>
  </plan>
```

When current_hour is less than 12, how many times do I have to call 'ExecutePlan' until the plan is complete? Once or twice?

Reply:
2 Times.
1st will process the IF.
2nd will process the function.